server/chat-commands/admin.ts

```diff
@@ -389,14 +395,17 @@ export const commands: ChatCommands = {
 		await rebuild(this);
 
 		const lock = Monitor.hotpatchLock;
-		const hotpatches = ['chat', 'formats', 'loginserver', 'punishments', 'dnsbl', 'modlog'];
+		const hotpatches = [
+			'chat', 'formats', 'loginserver', 'punishments', 'dnsbl', 'modlog',
+			'processmanager', 'roomsp', 'usersp',
+		];
+		if (lock[target]) {
+			return this.errorReply(`That hotpatch has been disabled by ${lock[target].by} (${lock[target].reason})`);
+		}
```

Comment:
Certain hotpatch commands have aliases (like `/hotpatch commands` → `chat`) that you haven't accounted for, and even the ones you did account for (like `battles`), the approach seems very bug-prone. And while we've merged hotpatching formats/battles, I don't think it's unthinkable that we will have hotpatch commands that combine other hotpatch commands together, which this approach doesn't support well.

Reply:
Ok, I can restore that.

Reply:
Done.